subsys/net/ip/dhcpv4.c

```diff
@@ -745,6 +850,46 @@ static enum net_verdict parse_options(struct net_if *iface,
 			net_if_ipv4_set_gw(iface, &router);
 			break;
 		}
+		case DHCPV4_OPTIONS_DNS_SERVER: {
+			const unsigned int max_ipv4_addr_len = 16;
+			struct in_addr dns;
+			char dns_string[max_ipv4_addr_len];
```

Comment:
Use NET_IPV4_ADDR_LEN here, then you can also remove max_ipv4_addr_len variable.